client/src/store/users/reducers.ts

```diff
@@ -0,0 +1,53 @@
+import {UserState} from '../types';
+import {LOGIN, LOGOUT, User, UserActionTypes} from './types';
+
+const initialUsers: User[] = [
+  {
+    name: 'Siddhartha Gupta',
+    userName: "siddhero97",
+    isLoggedIn: false
+  },
+  {
+    name: 'Kwan Lam',
+    userName: "masterKwan",
+    isLoggedIn: false
+  }
+];
+
+export const initialState: UserState = {
+  users: initialUsers,
+  isLoggedIn: false,
+  currUser: null,
+};
+function isLoggedIn(users: User[]): boolean {
+  let isLoggedIn = false;
+  for(const user of users) {
```

Comment:
This is a perfect example of why we should not have an array of users in the redux store. We probably don't want to loop through 100k users on the client side. We can remove this reducer.